docs/Getting-Started.md

```diff
@@ -56,61 +44,167 @@ fastify.listen(3000, function (err) {
 
-fastify.register(require('./route'))
+fastify.register(require('./our-first-route'))
 
-const opts = {
-  hello: 'world',
-  something: true
+fastify.listen(3000, function (err) {
+  if (err) throw err
+  console.log(`server listening on ${fastify.server.address().port}`)
+})
+```
+
+```js
+// our-first-route.js
+
+async function routes (fastify, options) {
+  fastify.get('/', async (request, reply) => {
+    return { hello: 'world' }
+  })
 }
 
-fastify.register([
-  require('./another-route'),
-  require('./yet-another-route')
-], opts, function (err) {
-  if (err) throw err
+module.exports = routes
+```
+In this we used the `register` API. This API is the core of the Fastify framework, and is the only way to register routes, plugins and so on. 
+
+At the beginning of this document we said that Fastify helps you on handling the asynchronous bootstrapping of your application. Why this is important?  
+Well, let's say you must connect to database to fetch some data, obviously you don't want that the server starts accept requests before the database connection has been established. How can you address this problem?  
+Usually you will work with some complex callback/promises system that will mix the framework api with other libraries and your code.  
+Fastify handles this internally for you, with the minimum effort!
+
+Let's rewrite the above example with a database connection.  
+*(now we'll use a simple example, if you want a robust sulution you can use [`fastify-mongo`](https://github.com/fastify/fastify-mongodb) or checkout our [ecosystem](https://github.com/fastify/fastify/blob/master/docs/Ecosystem.md))*
+```js
+const fastify = require('fastify')()
+
+fastify.register(require('./our-db-connector'), {
+  url: 'mongodb://mongo/db'
 })
+fastify.register(require('./our-first-route'))
 
-fastify.listen(8000, function (err) {
+fastify.listen(3000, function (err) {
   if (err) throw err
   console.log(`server listening on ${fastify.server.address().port}`)
 })
 ```
 
 ```js
-/* route.js */
+// our-db-connector.js
+const MongoClient = require('mongodb').MongoClient
 
-module.exports = function (fastify, options, next) {
-  fastify.get('/', function (req, reply) {
-    reply.send({ hello: 'world' })
-  })
-  next()
+async function db (fastify, options) {
+  const url = options.url
+  delete options.url
+
+  const db = await MongoClient.connect(url, options)
+  fastify.decorate('mongo', db)
 }
+
+module.exports = db
 ```
 
-or with `async/await`:
 ```js
-/* route.js */
+// our-first-route.js
+
+async function routes (fastify, options) {
+  const collection = fastify.mongo.collection('test')
 
-module.exports = async function (fastify, options) {
-  fastify.get('/', function (req, reply) {
-    reply.send({ hello: 'world' })
+  fastify.get('/', async (request, reply) => {
+    return { hello: 'world' }
+  })
+
+  fastify.get('/search/:id', async (request, reply) => {
+    return await collection.findOne({ id: req.params.id })
```

Comment:
If the query doesn't return any document, which is the behaviour? I know that this is an example but maybe:
```
const Boom = require('boom')
fastify.get('/search/:id', async (request, reply) => {
  const doc = await collection.findOne({ id: req.params.id })
  if (!doc) throw Boom.notFound()
  return doc
})
```

It easies to introduce the build-in error handing.

Reply:
This is a very basic example, I don't want to make it more complex than needed. Here we are just showing the async/await support and how use a db connector with decorate.

Reply:
I'm not a big fan of "incomplete" examples. Anyway I understand this is out of the scope of "Your first plugin" section. But I'd like to have a section in this file that describes the error handling: it is a great feature!

Reply:
> I'm not a big fan of "incomplete" examples

Me too, but your proposal is out of scope. If you show the usage of Boom in a getting started example a user will probably think that is required, which is completely false.

The error handling is described in the plugin guide.